src/node_native_module.cc

```diff
@@ -174,6 +174,64 @@ MaybeLocal<Function> NativeModuleLoader::CompileAsModule(
   return LookupAndCompile(context, id, &parameters, result);
 }
 
+#ifdef NODE_BUILTIN_MODULES_PATH
+static std::string OnDiskFileName(const char* id) {
+  std::string filename = NODE_BUILTIN_MODULES_PATH;
+  filename += "/";
+
+  if (memcmp(id, "internal/deps", strlen("internal/deps")) == 0) {
```

Comment:
This is not safe when `strlen(id) < strlen("internal/deps")`. You want `strcmp()` or `strncmp()`.